invisible_cities/reco/deconv_functions.py

```diff
@@ -0,0 +1,288 @@
+        pass_df._is_copy = False
+        with np.errstate(divide='ignore'):
+            for redist_variable in variables:
+                pass_df[redist_variable] = pass_df[redist_variable] * np.divide(df[redist_variable].sum(), pass_df[redist_variable].sum())
+        return pass_df
+
+    return cut_and_redistribute
+
+def drop_isolated_sensors(distance  : List[float]=[10., 10.],
+                          variables : List[str  ]=[        ]) -> Callable:
+    """
+    Drops rogue/isolated hits (SiPMs) from a groupedby dataframe.
+
+    Parameters
+    ----------
+    df      : GroupBy ('event' and 'npeak') dataframe
+
+    Initialization parameters:
+        distance  : Distance to check for other sensros. Usually equal to sensor pitch.
+        variables : List with variables to be redistributed.
+
+    Returns
+    ----------
+    pass_df : hits after removing isolated hits
+    """
+    dist = np.sqrt(distance[0] ** 2 + distance[1] ** 2)
+
+    def drop_isolated_sensors(df : pd.DataFrame) -> pd.DataFrame:
+        x       = df.X.values
+        y       = df.Y.values
+        mask_xy = [False if np.ma.masked_equal(np.sqrt((x - xi)**2 + (y - yi)**2), 0.0, copy=False).min() > dist
+                         else True
+                         for xi, yi in zip(x, y)]
+        pass_df = df[mask_xy]
+        pass_df._is_copy = False
+        with np.errstate(divide='ignore'):
+            for redist_variable in variables:
+                pass_df[f'{redist_variable}'] = pass_df[redist_variable] * np.divide(df[redist_variable].sum(), pass_df[redist_variable].sum())
+        return pass_df
+
+    return drop_isolated_sensors
+
+
+def deconvolutionInput(sampleWidth : List[float],
+                       bin_size    : List[float],
+                       interMethod : Optional[str]=None
+                       ) -> Callable:
+    """
+    Prepares the given data for deconvolution. This involves interpolation of
+    the data.
+
+    Parameters
+    ----------
+    data        : Sensor (hits) position points.
+    weight      : Sensor charge for each point.
+
+    Initialization parameters:
+        sampleWidth : Sampling size of the sensors.
+        bin_size    : Size of the interpolated bins.
+        interMethod : Interpolation method.
+
+    Returns
+    ----------
+    Hs          : Charge input for deconvolution.
+    interPoints : Coordinates of the deconvolution input.
+    """
+
+    def deconvolutionInput(data        : Tuple[np.ndarray, ...],
+                           weight      : np.ndarray
+                           ) -> Tuple[np.ndarray, Tuple[np.ndarray, ...]]:
+        ranges  = [[data[i].min() - 1.5 * sw, data[i].max() + 1.5 * sw] for i, sw   in enumerate(sampleWidth)]
+        allbins = [int(np.diff(rang)/sampleWidth[i])                    for i, rang in enumerate(     ranges)]
+        nbin    = [int(np.diff(rang)/bin_size[i])                       for i, rang in enumerate(     ranges)]
```

Comment:
This is kind of delicate, so... in reality you are right but this is a 'patchy' solution to some problem related of the uneven distribution of the Z associated to the slices. So... since slices are not separated equally in 2 us (at least the stored Z value) but distributed around 2 (but actually usually lower, as we talked at some point) I had several 0s when trying to interpolate. I decided on int so it would make bins lower than the rebinning in order to avoid 0s in the histogram as much as possible. 

However, I may try to just use uneven binning since that is a possibility, that should also work and remove all kind of binning problems.